resources/providers/azurelib/inventory/provider.go

```diff
@@ -64,28 +68,65 @@ type ProviderInitializerAPI interface {
 }
 
 func (p *ProviderInitializer) Init(ctx context.Context, log *logp.Logger, azureConfig auth.AzureFactoryConfig) (ServiceAPI, error) {
+	log = log.Named("azure")
+
 	clientFactory, err := armresourcegraph.NewClientFactory(azureConfig.Credentials, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	client := clientFactory.NewClient()
 
-	// We wrap the client so we can mock it in tests
+	// We wrap the client, so we can mock it in tests
 	wrapper := &AzureClientWrapper{
 		AssetQuery: func(ctx context.Context, query armresourcegraph.QueryRequest, options *armresourcegraph.ClientResourcesOptions) (armresourcegraph.ClientResourcesResponse, error) {
 			return client.Resources(ctx, query, options)
 		},
 	}
 
+	subscriptions, err := p.getSubscriptionIds(ctx, azureConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get subscription ids: %w", err)
+	}
+	log.Info(
+		lo.Reduce(subscriptions, func(agg string, item *string, _ int) string {
+			return fmt.Sprintf("%s %s", agg, strings.Dereference(item))
+		}, "subscriptions:"),
+	)
+
 	return &Provider{
-		Config: azureConfig,
-		client: wrapper,
-		log:    log,
-		ctx:    ctx,
+		log:           log,
+		client:        wrapper,
+		subscriptions: subscriptions,
+		ctx:           ctx,
+		Config:        azureConfig,
 	}, nil
 }
 
+func (p *ProviderInitializer) getSubscriptionIds(ctx context.Context, azureConfig auth.AzureFactoryConfig) ([]*string, error) {
+	// TODO: mockable
```

Comment:
`Init()` is not tested so I consider it out of scope to test this function for now, we should handle this technical debt after the feature freeze though